### PR TITLE
Add new `Style/SwapValues` cop

### PR DESCRIPTION
**Pull request by fatkodima**
This cop enforces the use of shorthand-style swapping of 2 variables.
```ruby
# bad
tmp = x
x = y
y = tmp

# good
x, y = y, x
```

This will be mostly useful for newcomers from other languages into ruby, because when you used it once you will always remember about its existence.

### CODE REVIEW

**Comment by bbatsov** [comment]
Great idea! Seems that we should also add this to the style guide.

File: docs/modules/ROOT/pages/cops_style.adoc [hunk 1]
```diff
@@ -9897,6 +9897,35 @@ end
...
+
+| Pending
+| Yes
+| Yes (Unsafe)
+| 1.1
+| -
+|===
+
+This cop enforces the use of shorthand-style swapping of 2 variables.
+It is marked as unsafe, because while autocorrecting it can erroneously remove
```
Comment (bbatsov) [inline docs/modules/ROOT/pages/cops_style.adoc:9913]: The wording here is a bit misleading, as the cop is safe and only the auto-correct is unsafe.

**Comment by fatkodima** [comment]
Added link to the style guide (https://github.com/rubocop-hq/ruby-style-guide/pull/853) and changed wording.

**Comment by bbatsov** [comment]
Thanks!

**Comment by pvdb** [comment]
There's an issue #10394 with this cop that I just ran into.

**Pull request merged**